src/devices/DCMotor/DCMotor2Pin.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device.DCMotor
+{
+    internal class DCMotor2Pin : DCMotor
+    {
+        private PwmChannel _pwm;
+        private int _DIRpin;
+        private double _speed;
+
+        public DCMotor2Pin(
+            PwmChannel pwmChannel,
+            int pin1,
+            GpioController controller,
+            bool shouldDispose)
+            : base(controller ?? ((pin1 == -1) ? null : new GpioController()), controller == null ? true : shouldDispose)
+        {
+            _pwm = pwmChannel;
+
+            _DIRpin = pin1;
+
+            _speed = 0;
+
+            _pwm.Start();
+
+            if (_DIRpin != -1)
```

Comment:
I'd say you should throw if it's not defined since we already have a class which handles that specific case where this is `-1`. You cannot have bi-directional pin and not have it at the same time